neuralprophet/components/base.py

```diff
@@ -0,0 +1,35 @@
+import torch
+import torch.nn as nn
+
+
+class BaseComponent(nn.Module):
+    def __init__(self, n_forecasts, quantiles, id_list, device):
+        super().__init__()
+        self.n_forecasts = n_forecasts
+        self.quantiles = quantiles
+        self.id_list = id_list
+        self.device = device
+
+    def forward(self, x):
```

Comment:
`@abstract`

Reply:
alternative to having separate Global/Local/Glocal classes:
introduce additional arg
```
if self.glocal == "global":
    return forward_global()
elif ...
```